File: 3364-minimum-positive-sum-subarray/3364-minimum-positive-sum-subarray.js
/**
 * @param {number[]} nums
 * @param {number} l
 * @param {number} r
 * @return {number}
 */
var minimumSumSubarray = function (nums, l, r) {
    if (nums.length < l) return -1;
    let sum = 0;
    let res = Number.MAX_VALUE;
    let n = nums.length;
    for (let i = 0; i < n; i++) {
        let m = Math.min(n, i + r);
        sum = 0;
        for (let j = i; j < m; j++) {
            sum += nums[j];
            if (j - i + 1 >= l && j - i + 1 <= r && sum > 0) {
                //console.log(sum)
                res = Math.min(res, sum);
            }
        }
    }
    if (res === Number.MAX_VALUE) res = -1;
    return res;
};

File: 3364-minimum-positive-sum-subarray/3364-minimum-positive-sum-subarray.java
class Solution {
    public int minimumSumSubarray(List<Integer> nums, int l, int r) {
        int size = nums.size(), res = -1;
	for(int s=l; s<=r; s++) {
		for(int i=0; i<=size-s; i++) {
			int sum = 0;
			for(int j=i; j<i+s; j++) {
				sum += nums.get(j);
			}
			if(sum > 0) {
				if(res == -1 || res > sum) res = sum;
			}
		}
	}
	return res;
    }
}